power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -750,6 +889,57 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
         return tap_changed;
     }
 
+    bool adjust_transformer_bs(RegulatedTransformer const& regulator, State const& state,
+                               ResultType const& solver_output, UpdateBuffer& update_data,
+                               BinarySearchOptions const& bs_options) const {
+        auto const strategy_max = bs_options.strategy_max;
+        bool tap_changed = false;
+        auto& bs_ref = binary_search_[bs_options.idx_bs.x][bs_options.idx_bs.y];
+
+        regulator.transformer.apply([&](transformer_c auto const& transformer) { // NOSONAR
+            using TransformerType = std::remove_cvref_t<decltype(transformer)>;
+            using sym = typename ResultType::value_type::sym;
+
+            auto const param = regulator.regulator.get().template calc_param<sym>();
+            auto const node_state =
+                NodeState<sym>{.u = u_pu_controlled_node<TransformerType>(regulator, state, solver_output),
+                               .i = i_pu_controlled_node<TransformerType>(regulator, state, solver_output)};
+
+            if (bs_ref.get_end_of_bs() || bs_ref.get_bs_inevitable_run()) {
+                tap_changed = false;
+                return;
+            }
+
+            auto const cmp = node_state <=> param;
+            if (auto new_tap_pos =
+                    [&cmp, strategy_max, &bs_ref] {
+                        if (cmp != 0) {                                    // NOLINT(modernize-use-nullptr)
+                            bs_ref.propose_new_pos(strategy_max, cmp > 0); // NOLINT(modernize-use-nullptr)
+                        }
+                        return bs_ref.get_bs_current_tap();
+                    }();
+                new_tap_pos != transformer.tap_pos()) {
+                bs_ref.set_bs_current_tap(new_tap_pos);
+                add_tap_pos_update(new_tap_pos, transformer, update_data);
+                tap_changed = true;
+                return;
+            }
+
+            if (strategy_ == OptimizerStrategy::fast_any) {
+                tap_changed = false;
+                return;
+            }
+
+            const bool previous_down = bs_ref.get_bs_last_down();
+            bs_ref.recalibrate(strategy_max);
+
+            const auto tap_pos = bs_ref.post_process(strategy_max, previous_down, tap_changed);
```

Comment:
idem in other places
```suggestion
            bool const previous_down = bs_ref.get_bs_last_down();
            bs_ref.recalibrate(strategy_max);

            auto const tap_pos = bs_ref.post_process(strategy_max, previous_down, tap_changed);
```

Reply:
comment addressed in the latest commit.